### PR TITLE
add request_dir for allowing the url to be different then that of the storage path. 

### PR DESCRIPTION
**Pull request by ghost**
This is for allowing the url  to be different then that of the storage path.

### CODE REVIEW

**Comment by digitalcake** [comment]
The point of this was to allow me to set the storage path differently for my uploader via
```
  # Override the storage directory:
  def storage_dir(version, {file, scope}) do
    "priv/uploads/photos"
  end

  def request_dir(version, {file, scope}) do
    "uploads/photos"
  end
```

This allowed me to use plug static via
```
  plug Plug.Static,
    at: "/uploads/photos", from: {:digitalcakes, "priv/uploads/photos"}, gzip: false
```

**Comment by digitalcake** [comment]
ouch, welp it was worth a try... I guess.

File: lib/arc/storage/local.ex [hunk 1]
```diff
@@ -36,4 +36,11 @@ defmodule Arc.Storage.Local do
       Arc.Definition.Versioning.resolve_file_name(definition, version, file_and_scope)
     ])
   end
+
+  defp build_local_request_path(definition, version, file_and_scope) do
```
Comment (pedep) [inline lib/arc/storage/local.ex]: The `local` in this function name feels redundant. Personally, i would drop it from the name, but its not of huge importance

File: lib/arc/definition/storage.ex [hunk 2]
```diff
@@ -5,13 +5,14 @@ defmodule Arc.Definition.Storage do
       @async true
 
       def filename(_, {file, _}), do: Path.basename(file.file_name, Path.extname(file.file_name))
-      def storage_dir(_, _), do: "uploads"
+      def storage_dir(_, _), do: "priv/uploads"
```
Comment (pedep) [inline lib/arc/definition/storage.ex]: Not sure if i would change the default

**Review by pedep — COMMENTED** [review]
I think this PR is very close, but probably needs a few lines of doc to explain the usage.
I also would like to see a test verifying the url generated by arc matches the specified `request_dir`

**Comment by scarfacedeb** [comment]
Is it there any chance that this PR would be merged?

I basically reimplemented the same feature in my app. (+ asset_host for local storage). Should I create new PR with my changes too?

**Comment by pedep** [comment]
@scarfacedeb I hope so too. 
I can see we still need a spec to test the behavior. I might get around to writing that later this evening, but have some things at work to deal with first.

If you have something to contribute to whats already here, have a different / better implementation, or you would like to write the above mentioned spec, it would be very appreciated.
If you want to contribute to whats already here, i think you can just base your commits on this PR, and make a new PR that includes cakes commits.

**Comment by digitalcake** [comment]
Hello so sorry for the long reply. I will see if I can make sometime this week to update the pr. Thanks for the review.

**Comment by pedep** [comment]
@digitalcake No problem :smile: Your work is appreciated

**Comment by SLOBYYYY** [comment]
Why is this not merged yet? This is a problem I'm facing for a long time now. 7 months have passed and nothing happened so far.

Would you be please so kind and merge this?

**Comment by joshchernoff** [comment]
Sorry I've foolishly have deleted my local repo for this while cleaning up my account. 
I've also found using `ExAws.S3.presigned_url` a much more useful case for uploading files via the client and don't think I will be trying to use Arc in the future so sorry to say I'm gonna pass on this. 

I'll keep this open for anyone who wishes to try and finish what I started.

**Pull request closed without merging**